packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -27,8 +27,7 @@ navigatorIsReadyRef.current = false
 async function ensureNavigator() {
   let retries = 0
   while (
-    !navigationRef.current &&
-    !navigatorIsReadyRef.current &&
+    (!navigationRef.current || !navigatorIsReadyRef.current) &&
     retries < NAVIGATOR_INIT_RETRIES
   ) {
     await sleep(200)
```

Comment:
Should we fire an analytics event as well when this fail? Right below this there's an error thrown if the navigator is not initialized (`throw new Error('navigator is not initialized')`) maybe we can also send that to analytics to see how frequent this is.

Reply:
Good idea! I guess one thing to be careful with is making sure the analytics framework is initialized at that point but should be easy to test.